src/compiler/checker.ts

```diff
@@ -9257,6 +9316,12 @@ namespace ts {
                         if (!bodyDeclaration) {
                             bodyDeclaration = node;
                         }
+
+                        // abstract functions cannot have an implementation.
+                        // Extra checks are to avoid reporting multiple errors relating to the "abstractness" of the node.
+                        if (currentNodeFlags & NodeFlags.Abstract && node.kind === SyntaxKind.MethodDeclaration && !isConstructor) {

```

Comment:
That in itself doesn't seem like a problem. You could just error in such an invocation.

A better reason is that we don't allow statics to be marked as `abstract`, and since the constructor is part of the static side of a class, it should also be an error to mark it as abstract.